### PR TITLE
Update rollup and babel to latest versions

### PR DESCRIPTION
**Pull request by alex-page**
### WHY are these changes introduced?

Rollup and Babel are blocking our migration to minimum NodeJS version 16.

This PR updated all Rollup and Babel dependencies. These NPM package upgrades are not compatible with NodeJS 14.17 (main branch version). This change unblocks a larger change to our NodeJS versions.

### WHAT is this pull request doing?

- [x] Moves duplicate dependencies to root
- [x] Upgrades all rollup and babel dependencies
- [x] Fixes any required breaking changes or warnings
- [x] Compared diff of previous packages and new ones and it doesn't have a regression

### How to 🎩

1. Run a build on `main` save the built files on desktop
1. Run a build on this `bump-build-deps` save the built files on desktop
1. Run a diff over the built files
1. Make sure no regressions
1. CI, tests should pass and website should render all pages.

**@alex-page's output from Diff:** [diff.zip](https://github.com/Shopify/polaris/files/10531068/diff.zip)
```
diff -bur polaris-tokens/dist-main polaris-tokens/dist-v11 > icons.diff
diff -bur polaris-tokens/dist-main polaris-tokens/dist-v11 > tokens.diff
diff -bur polaris-migrator/dist-main polaris-migrator/dist-v11 > migrator.diff
diff -bur polaris-react/build-main polaris-react/build-v11 > react.diff
```

### 🎩 checklist

- [x] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [x] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [x] Updated the component's `README.md` with documentation changes
- [x] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: polaris-migrator/package.json [hunk 1]
```diff
@@ -48,10 +48,6 @@
   "devDependencies": {
     "@types/is-git-clean": "^1.1.0",
     "@types/jscodeshift": "^0.11.5",
-    "@rollup/plugin-babel": "^5.3.1",
-    "@rollup/plugin-commonjs": "^22.0.2",
-    "@rollup/plugin-json": "^4.1.0",
-    "@rollup/plugin-node-resolve": "^13.3.0",
```
Comment (alex-page) [inline polaris-migrator/package.json:54]: Moved to root

File: polaris-migrator/rollup.config.mjs [hunk 2]
```diff
@@ -21,14 +21,14 @@ export default {
   output: [
     {
       format: /** @type {const} */ ('cjs'),
-      entryFileNames: '[name][assetExtname].js',
```
Comment (alex-page) [inline polaris-migrator/rollup.config.mjs:24]: `[assetExtname]` no longer exists in the latest version of rollup.

> Note that in this case, `[name]` will include the relative path from the output root and possibly the original file extension if it was not one of `.js`, `.jsx`, `.mjs`, `.cjs`, `.ts`, `.tsx`, `.mts`, or `.cts`.
https://rollupjs.org/configuration-options/#output-entryfilenames

File: polaris-icons/rollup.config.mjs [hunk 3]
```diff
@@ -222,24 +222,26 @@ export default [
...
     ],
-    manualChunks: (id) => {
-      // Generate distinct chunks for each icon
-      // This allows consuming apps to split up the icons into multiple subchunks
-      // containing a few icons each instead of always having to put every icon
-      // into a single shared chunk
-      if (id.startsWith(iconBasePath)) {
-        return id.replace(iconBasePath, 'icons/');
-      }
-    },
```
Comment (alex-page) [inline polaris-icons/rollup.config.mjs:242]: `manualChunks` is going to move to `output` in the next breaking change. This moves it now so we don't have to fix it later.

**Comment by alex-page** [comment]
shopify/polaris-icons diff

```diff
diff -bur dist-main/index.js dist-v11/index.js
--- dist-main/index.js	2023-01-30 09:40:33
+++ dist-v11/index.js	2023-01-30 08:57:46
@@ -1,7 +1,5 @@
 'use strict';
 
-Object.defineProperty(exports, '__esModule', { value: true });
-
 var icons__AbandonedCartMajor_svg = require('./icons/AbandonedCartMajor.svg.js');
 var icons__AccessibilityMajor_svg = require('./icons/AccessibilityMajor.svg.js');
 var icons__ActivitiesMajor_svg = require('./icons/ActivitiesMajor.svg.js');
```

**Review by jesstelford — APPROVED** [review]

**Comment by alex-page** [comment]
I am going to merge this into v11. Please still take a look/review I will be happy to iterate in a follow up PR.

**Pull request merged**